src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2094,7 +2114,46 @@ public void HandleIDebugEntryPointEvent2(IDebugEngine2 pEngine, IDebugProcess2 p
 
         public void HandleIDebugBreakpointEvent2(IDebugEngine2 pEngine, IDebugProcess2 pProcess, IDebugProgram2 pProgram, IDebugThread2 pThread, IDebugEvent2 pEvent)
         {
-            FireStoppedEvent(pThread, StoppedEvent.ReasonValue.Breakpoint);
+            bool fireStopEvent = true;
+
+            if (pEvent is IDebugBreakpointEvent2 debugBreakpointEvent) {
+                debugBreakpointEvent.EnumBreakpoints(out IEnumDebugBoundBreakpoints2 pBoundBreakpoints);
+                IDebugBoundBreakpoint2[] boundBp = new IDebugBoundBreakpoint2[1];
+                uint numReturned = 0;
+                while (pBoundBreakpoints.Next(1, boundBp, ref numReturned) == HRConstants.S_OK && numReturned == 1)
+                {
+                    if (boundBp[0].GetPendingBreakpoint(out IDebugPendingBreakpoint2 ppPendingBreakpoint) == HRConstants.S_OK)
+                    {
+                        if (ppPendingBreakpoint.GetBreakpointRequest(out IDebugBreakpointRequest2 ppBPRequest) == HRConstants.S_OK)
+                        {
+                            if (ppBPRequest is AD7BreakPointRequest ad7BreakpointRequest)
+                            {
+                                if (m_tracepointManager.Contains(ad7BreakpointRequest.Id))
+                                {
+                                    fireStopEvent = false;
+
+                                    ThreadPool.QueueUserWorkItem((o) =>
+                                    {
```

Comment:
Do we need to kick this to the thread pool?

Reply:
Yes. EnumFrameInfo runs `_engine.DebuggedProcess.WorkerThread.RunOperation(async () => stackFrames = await _engine.DebuggedProcess.ThreadCache.StackFrames(_debuggedThread));` and `RunOperation` which has an assert `!IsPollThread()`. If it does not run in the thread pool, it deadlocks.